Battery_Checker.h

```diff
@@ -0,0 +1,106 @@
+({                                                                                                         \
+    int checkResult = (int)((currentValue >= bmsFactorRanges[(currentFactor*5)+LOW_BREACH][0])             \
+                           && (currentValue <= bmsFactorRanges[(currentFactor*5)+LOW_BREACH][1]));         \
+    checkResult;                                                                                           \
+})
+
+#define ISLOWLIMITWARNING(currentValue, currentFactor)                                                      \
+({                                                                                                          \
+    int checkResult = (int)((currentValue >= bmsFactorRanges[(currentFactor*5)+LOW_WARNING][0])             \
+                           && (currentValue <= bmsFactorRanges[(currentFactor*5)+LOW_WARNING][1]));         \
+    checkResult;                                                                                            \
+})
+
+#define ISNORMALLIMIT(currentValue, currentFactor)                                                          \
+({                                                                                                          \
+    int checkResult = (int)((currentValue >= bmsFactorRanges[(currentFactor*5)+NORMAL_STATE][0])            \
+                           && (currentValue <= bmsFactorRanges[(currentFactor*5)+NORMAL_STATE][1]));        \
+    checkResult;                                                                                            \
+})
+
+#define ISHIGHLIMITWARNING(currentValue, currentFactor)                                                     \
+({                                                                                                          \
+    int checkResult = (int)((currentValue >= bmsFactorRanges[(currentFactor*5)+HIGH_WARNING][0])            \
+                           && (currentValue <= bmsFactorRanges[(currentFactor*5)+HIGH_WARNING][1]));        \
+    checkResult;                                                                                            \
+})
+
+#define ISHIGHLIMITBREACH(currentValue, currentFactor)                                                      \
+({                                                                                                          \
+    int checkResult = (int)((currentValue >= bmsFactorRanges[(currentFactor*5)+HIGH_BREACH][0])             \
+                           && (currentValue <= bmsFactorRanges[(currentFactor*5)+HIGH_BREACH][1]));         \
+    checkResult;                                                                                            \
+})
+
+#define checkOK(currentValue, currentFactor)                       \
+({                                                                 \
+    int checkResult = OK;                                          \
```

Comment:
if low limit is breached , then rest of the conditions needn't be executed. In this case, no matter what the value is every if condition seems to be checked. Is this the case? if so then definitely it should be avoided.

Reply:
Yes, now i corrected it in code.